code/modules/reagents/reagent_containers/glass_containers.dm

```diff
@@ -117,11 +143,11 @@
 		return
 
 	else if(reagents.total_volume)
-		to_chat(user, "<span class='notice'>You splash the solution onto [target].</span>")
-		reagents.reaction(target, TOUCH)
-		spawn(5) reagents.clear_reagents()
-		return
-
+		if(user.a_intent == INTENT_HARM)
```

Comment:
This `if` clause seems like it should be bundled into the `else if` one line above